### PR TITLE
Implement transformation configuration via utilities

### PR DESCRIPTION
**Pull request by jsnmoon**
This addresses the FIXME's in several of the transforms. It spreads in jscodeshift's [options](https://github.com/facebook/jscodeshift#options) argument to a pre-defined default configuration inside the utility file.

### CODE REVIEW

File: utils/main.js [hunk 1]
```diff
@@ -162,6 +162,28 @@ var util = {
...
+  },
+
+  getConfig: function(options) {
+    var defaultConfig = util.getDefaultConfig();
+    var out = {};
+    var keys1 = Object.keys(defaultConfig);
+    for (var i = 0, key1 = keys1[i]; i < keys1.length; i++) {
+      out[key1] = defaultConfig[key1];
+    }
+    if (options !== null && options !== undefined && typeof options === 'object' && !Array.isArray(options)) {
```
Comment (gmathieu) [inline utils/main.js]: `typeof options === 'object'` should cover the first two falsy checks (null and undefined)

Comment (jsnmoon) [inline utils/main.js]: I think you're right on the undefined check as being unnecessary, but null is actually... an object. ¯\_(ツ)_/¯

`typeof null === 'object' === true`

**Comment by jsnmoon** [comment]
Right on! I'll get to improving the README with this change.

File: utils/main.js (hunk 1 above)
Comment (gmathieu) [inline utils/main.js]: Oh man... This reminds me of [wat](https://www.destroyallsoftware.com/talks/wat). Thanks for verifying.

**Comment by jsnmoon** [comment]
I made some further modifications to allow the config to distinguish between 5to6-codemod configuration and Recast configuration. Also updated the readme!

File: utils/main.js [hunk 2]
```diff
@@ -162,8 +162,78 @@ var util = {
...
+    }
+    var out = {};
+    var keys1 = Object.keys(defaultConfig);
+    for (var i = 0; i < keys1.length; i++) {
+      var key1 = keys1[i];
+      if (keyFilterFunction(key1)) {
+        out[key1] = defaultConfig[key1];
+      }
+    }
+    // WAT: typeof undefined === 'object' === true
```
Comment (gmathieu) [inline utils/main.js]: typeof null?

Comment (jsnmoon) [inline utils/main.js]: D'oh... you're right.

**Comment by jsnmoon** [comment]
@xjamundx: what do you think?

**Comment by jsnmoon** [comment]
@gmathieu: Could you possibly push my fork's feature branch onto a feature branch here? I would like to open a new PR for #34 for review.

File: package.json [hunk 3]
```diff
@@ -17,6 +17,7 @@
   },
   "devDependencies": {
     "eslint": "^1.0.0",
+    "lodash.assign": "^4.2.0",
```
Comment (gmathieu) [inline package.json]: Wouldn't hurt to have all of lodash available. It's not like we're trying optimize on size since this server side JS.

Comment (jsnmoon) [inline package.json]: Got it - added `lodash` to `dependencies` and removed `lodash.assign` from `devDependencies`.

File: test/utils.js [hunk 4]
```diff
@@ -5,7 +5,7 @@
  */
 var recast = require('recast');
 var assert = require('assert');
-var assign = require('lodash.assign');
+var assign = require('lodash/assign');
```
Comment (jsnmoon) [inline test/utils.js]: Ha, whoops! Thanks for the catch.

**Pull request merged**